scripts/auto_archive.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import annotations
+
+import argparse
+import json
+import logging
+import os
+from datetime import datetime
+from datetime import timedelta
+
+import git
+import humanfriendly
+from datalad.plugin import export_archive
+from github import Github
+
+from scripts.datalad_utils import get_dataset
+from scripts.datalad_utils import install_dataset
+from scripts.log import get_logger
+from tests.functions import get_proper_submodules
+
+
+logger = get_logger(
+    "CONP-Archive", filename="conp-archive.log", file_level=logging.DEBUG
+)
+
+
+class ArchiveFailed(Exception):
+    pass
+
+
+def parse_args():
+    example_text = """Example:
+    PYTHONPATH=$PWD python scripts/auto_archive.py <out_dir>
+    """
+
+    parser = argparse.ArgumentParser(
+        description="Archiver for the CONP-datasets.",
+        epilog=example_text,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
+        "out_dir", type=str, help="Path to store the archived datasets."
+    )
+    parser.add_argument(
+        "--all",
+        action="store_true",
+        help="Archive all the datasets rather than those modified since the last time.",
+    )
+    parser.add_argument(
+        "--max-size",
+        type=float,
+        default=20.0,
+        help="Maximum size of dataset to archive in GB.",
+    )
+
+    return parser.parse_args()
+
+
+def get_all_datasets():
+    return {submodule.path for submodule in git.Repo().submodules}
+
+
+def get_modified_datasets(
+    *,
+    since: datetime | None = None,
+    until: datetime | None = None,
+) -> set[str]:
+    """Retrieve the modified datasets.
+
+    Requires to set GITHUB_ACCESS_TOKEN as an environment variable.
+
+    Parameters
+    ----------
+    since : Optional[datetime], optional
+        Start date from which commits are retrieved, by default date of the previous crawl, if never crawled set to
+        one week ago.
+    until : Optional[datetime], optional
+        Latest date at which commit are retrieved, by default `now`
+
+    Returns
+    -------
+    set[str]
+        Path of the dataset folders.
+    """
+    now = datetime.now().astimezone()
+
+    if since is None:
+        if os.path.exists(".conp-archive"):
+            with open(".conp-archive") as fin:
+                since = datetime.fromisoformat(fin.read())
+        else:
+            since = now - timedelta(weeks=1)
+
+    if until is None:
+        until = now
+
+    try:
+        gh_access_token = os.environ.get("GITHUB_ACCESS_TOKEN", None)
+        if gh_access_token is None:
+            raise OSError("GITHUB_ACCESS_TOKEN is not defined.")
```

Comment:
I have defined the GITHUB_ACCESS_TOKEN variable but I always get the following error:
```
(py39_conpdataset) cecile@cecile-dev:.../conp-dataset$ PYTHONPATH=$PWD python scripts/auto_archive.py /data-scratch1/test/out_archive/
CONP-Archive - CRITICAL - GITHUB_ACCESS_TOKEN is not defined.
```

It is in my .bashrc and if I type `echo $GITHUB_ACCESS_TOKEN`, the token is indeed properly saved in the variable. What am I doing wrong?

Reply:
Was the variable `GITHUB_ACCESS_TOKEN` exported after defining it?

Reply:
aaaaaah, I forgot to add export in front of the variable name in my `.bashrc`... :( 
Thanks!